src/schema/schema_alter.c

```diff
@@ -250,9 +250,18 @@ __wt_schema_alter(WT_SESSION_IMPL *session,
     const char *uri, const char *newcfg[])
 {
 	WT_DECL_RET;
+	WT_SESSION *wt_session;
+	WT_SESSION_IMPL *int_session;
 
-	WT_RET(__wt_meta_track_on(session));
-	ret = __schema_alter(session, uri, newcfg);
-	WT_TRET(__wt_meta_track_off(session, true, ret != 0));
+	WT_RET(__wt_schema_internal_session(session, &int_session));
+	if (int_session != session)
+		wt_session = &int_session->iface;
+	else
+		wt_session = NULL;
+	WT_ERR(__wt_meta_track_on(int_session));
+	ret = __schema_alter(int_session, uri, newcfg);
+	WT_TRET(__wt_meta_track_off(int_session, true, ret != 0));
+err:	if (wt_session != NULL)
+		WT_TRET(wt_session->close(wt_session, NULL));
```

Comment:
Consider a `__wt_schema_session_release(session, int_session)` function that does the close if the two session pointers differ.  I think that would simplify all of the calling code and avoid the repeated `wt_session` dance.

Reply:
That is a good cleanup. I'll make that change.